### PR TITLE
Feat: Featured and products data to context.

### PR DESCRIPTION
**Pull request by zoef-yt**
Changed how I was getting data(faker.js) to useContext(mockbee)

# Files to check 

- [x] [data-context.jsx](https://github.com/zoef-yt/react-ecom/compare/changing-suggestion?expand=1#diff-cd61fac896071b874519080fc992463f910afbf8fd1dd1c3bc3875140d18c588)
- [x] [homepage-body.jsx](https://github.com/zoef-yt/react-ecom/compare/changing-suggestion?expand=1#diff-bf40967058d6a9afe861425cb54a466dae4dd154ebc265c9962bf2b0fe206c8e)

# Things to not check

- [ ] Functionality of the app

[Preview link]()

### CODE REVIEW

**Review by iRohitGaur — APPROVED** [review]
UI looks good. Code is understandable. Nice work! Good to merge 👍🏻

File: src/context/data/data-context.jsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+	}, []);
+
+	const getData = async () => {
+		try {
+			setProducts((prev) => ({ ...prev, loading: true }));
+			setFeaturedProducts((prev) => ({ ...prev, loading: true }));
+			const response = await axios.get('/api/products');
+			const data = [...response.data.products];
+			setProducts((prev) => ({ ...prev, data: data, loading: false }));
+			const featured = [...data].filter((product, index) => (index < 10 ? (Math.random() >= 0.4 ? product : null) : product));
```
Comment (sudipkundu999) [inline src/context/data/data-context.jsx:19]: Appreciation: It's good to see randomized featured products everytime

**Review by sudipkundu999 — APPROVED** [review]
Clean and easy to understand code. Good work.

File: src/context/data/data-context.jsx (hunk 1 above)
Comment (zoef-yt) [reply src/context/data/data-context.jsx:19]: Thanks. ❤️

**Pull request merged**